docs: Add resource and controller relationship matrix  txt

kahirokunn opened the pull request:
## Overview
This PR adds a reference page showing the relationship between Sveltos resources and their managing/creating controllers.

- New file: `docs/reference/resource-owners.md` - Resource and controller relationship matrix
- Updated `mkdocs.yml`: Added new "Reference" section

gianlucam76 commented:
Thank you @kahirokunn 

I will finish a new release this week (with few enhancements and bugs) and then will help with this. it is a great initiative. thank you

kahirokunn commented:
@gianlucam76
Thank you for your kind feedback! I appreciate your work on the upcoming release with enhancements and bug fixes.
I'm glad you find this documentation initiative valuable. I hope it helps more people understand and use the Sveltos project more effectively.
Looking forward to continuing to contribute to the project!
Thanks again!

gianlucam76 commented:
@kahirokunn  here is a summary:

- SveltosCluster controller manages SveltosCluster instances. For evert SveltosCluster, periodically connects to cluster and runs readiness and liveness checks.
- addon-controller manages ClusterProfile/Profile. For every cluster matching a profile, a ClusterSummary is created. All resources listed in the clusterSummary are deployed to the managed cluster.
- classifier deploys Classifier instances to managed clusters. It is also in charge of deploying sveltos-agent to managed clusters. This is done when the first classifier instance is deployed to managed cluster. Classifier also considers the classfierReport instances generated by sveltos-agent and update cluster labels accordingly
- event-manager manages EventTrigger instances. Deploys EventSource instances to managed cluster and process EventReports generated by sveltos-agent. Processing EventReports, new ClusterProfiles are created by event-manager
- healthcheck-manager manages ClusterHealthChecks. Deploys HealthChecks to managed clusters and processes the HealthReport generated by sveltos-agent. Send notifications based on  ClusterHealthCheck configurations
- techsupport manages Techsupport instances. Collect logs/events/resources from managed cluster and management cluster. Delivers messages with collected tech support
- shard-controller is in charge of deploying (and undeploying) new instances of Sveltos when clusters are marked as belonging to different shards

kahirokunn commented:
@gianlucam76,

Thank you for your review! Based on your feedback, I have clarified the relationships between controllers and resources and added a new "Controller Overview" section. Additionally, I have reorganized resource creation patterns as you suggested.

Furthermore, I've added columns to explicitly indicate where resources are created for both Local Agent Mode and Centralised Agent Mode (Agentless Mode).

However, as I have not personally used Agent Mode yet, there might be inaccuracies especially around these areas. If possible, could you please review these sections again and let me know if you find any inaccuracies or points for improvement?

Thank you again for your assistance! 🙏

egrosdou01 commented on docs/reference/resource-owners.md:
```diff
@@ -0,0 +1,71 @@
+---
+title: Sveltos Resources and Controller Map
```
@kahirokunn feel free to include your username as a writer. Like the below example.

```
authors:
    - Kahirokunn
```
